website/docs/r/cdn_frontdoor_custom_domain_association.html.markdown

```diff
@@ -13,9 +13,100 @@ Manages the association between a Front Door (standard/premium) Custom Domain an
 ## Example Usage
 
 ```hcl
-resource "azurerm_cdn_frontdoor_custom_domain_association" "example" {
+resource "azurerm_resource_group" "example" {
+  name     = "example-cdn-frontdoor"
+  location = "West Europe"
+}
+
+resource "azurerm_dns_zone" "example" {
+  name                = "domain.com"
+  resource_group_name = azurerm_resource_group.example.name
+}
+
+resource "azurerm_cdn_frontdoor_profile" "example" {
+  name                = "example-profile"
+  resource_group_name = azurerm_resource_group.example.name
+  sku_name            = "Standard_AzureFrontDoor"
+}
+
+resource "azurerm_cdn_frontdoor_origin_group" "example" {
+  name                     = "example-origin-group"
+  cdn_frontdoor_profile_id = azurerm_cdn_frontdoor_profile.example.id
+  session_affinity_enabled = true
+
+  restore_traffic_time_to_healed_or_new_endpoint_in_minutes = 10
+
+  health_probe {
+    interval_in_seconds = 240
+    path                = "/healthProbe"
+    protocol            = "Https"
+    request_type        = "HEAD"
+  }
+
+  load_balancing {
+    additional_latency_in_milliseconds = 0
+    sample_size                        = 16
+    successful_samples_required        = 3
+  }
+}
+
+resource "azurerm_cdn_frontdoor_origin" "example" {
+  name                          = "example-origin"
+  cdn_frontdoor_origin_group_id = azurerm_cdn_frontdoor_origin_group.example.id
+  enabled                       = true
+
+  certificate_name_check_enabled = false
+
+  host_name          = "contoso.com"
+  http_port          = 80
+  https_port         = 443
+  origin_host_header = "www.contoso.com"
+  priority           = 1
+  weight             = 1
+}
+
+resource "azurerm_cdn_frontdoor_endpoint" "example" {
+  name                     = "example-endpoint"
+  cdn_frontdoor_profile_id = azurerm_cdn_frontdoor_profile.example.id
+}
+
+resource "azurerm_cdn_frontdoor_rule_set" "example" {
+  name                     = "ExampleRuleSet"
+  cdn_frontdoor_profile_id = azurerm_cdn_frontdoor_profile.example.id
+}
+
+resource "azurerm_cdn_frontdoor_route" "contoso" {
```

Comment:
```suggestion
resource "azurerm_cdn_frontdoor_route" "example" {
```

Reply:
I did that intentionally to show the relationship between the `azurerm_cdn_frontdoor_route`, `azurerm_cdn_frontdoor_custom_domain` and the `azurerm_cdn_frontdoor_custom_domain_association` resources. That said I am happy to switch that around to make the documentation consistent. 🙂